taf/git.py

```diff
@@ -101,9 +104,14 @@ def __init__(
         self.default_branch = default_branch
 
     _pygit = None
-
+    
     @property
     def pygit(self):
+        if not self.is_git_repository:
+            raise GitError(
+                self,
+                message=f"The path '{self.path}' is not a Git repository.",
+            )
```

Comment:
If this error is not raised, but we are still not able to instantiate the pygit repo, raise `PygitError`. There is a `try - expect` here.
``` try:
       self._pygit = PyGitRepository(self)
    except Exception:
        pass
```
raise the error inside that `except Exception` - instead of "swallowing" it.